### ISSUE TITLE
Swallowing exceptions with logging

### ISSUE THREAD

**Issue by leotsarev** [SergeyTeplyakov/ExceptionAnalyzer#27]
Consider following scenario:
```
catch (SomethingException ex) {
   log.Error(ex);
   return false;
}
```
Why do you mark this as error? If exception already observed & logged, why this is a smell?

**Comment by SeriousM** [comment]
I was about to ask the same question.
It's completely fine to handle exceptions like that.

**Comment by SergeyTeplyakov** [comment]
Nope, this should be absolutely fine. I think there is a bug somewhere. THe idea is to warn for this case: 

```csharp
catch (Exception ex) {
   log.Error(ex.Message);
   return false;
}
```

In ErrorProne.NET I think I've loosen this rule and warn only when `ex` type is one of the base exception types and only when the "whole" exception is not observed, for instance, when only `ex.Message` is printed out.

**Comment by leotsarev** [comment]
For me, I think that case 
```catch (Exception ex) {
   log.Error(ex);
   return false;
}```
should not warn. It's pretty common pattern in our code - even this action completely failed, we should continue (probably act on other entities) and analyzer job to ensure that exception is logged.

**Comment by SergeyTeplyakov** [comment]
@leotsarev I totally agree with you:), in my version the error handler captured only `ex.Message` and in this case, the analyzer should emit a warning if `ex` is `System.Exception` because important information could be hidden inside an inner exception.

Here is an example:

```csharp
class WillThrow
{
    public WillThrow() => throw new System.Exception("Ooops");
    public static void Demo()
    {
        try {System.Activator.CreateInstance<WillThrow>();}
        catch(System.Exception e) {System.Console.WriteLine(e.Message); }
    }
}
```

In this case, you'll see only: "Exception has been thrown by the target of an invocation.", which is definitely not ok.

That's why ErrorProne.NET warns you if you catch `System.Excpetion` but not observe the "whole" exception.
